cabot/cabotapp/models.py

```diff
@@ -473,14 +481,43 @@ def last_result(self):
     def should_run(self):
         '''Returns true if the check should run, false otherwise.'''
 
-        # Do not run if the activity counter is enabled and zero.
-        # - If the DB entry does not exist, we assume a value of zero.
+        # Handle special cases for activity-counted checks, which may have run delays
         if self.use_activity_counter:
+            # If the counter value is zero, don't run
             counters = ActivityCounter.objects.filter(status_check=self)
             if len(counters) == 0 or counters[0].count <= 0:
                 logger.info("Skipping check '{}', activity counter is zero".format(self.name))
                 return False
 
+            counter = counters[0]
+
+            # last_enabled may be None when this change is first deployed. Set it to now and log a warning.
+            if counter.last_enabled is None:
+                counter.last_enabled = timezone.now()
+                counter.save()
```

Comment:
I think this could cause a race condition if a check is incremented/decremented as `should_run()` runs...could fix by doing the same `if is None` logic as `window_end` on `window_start`?

Reply:
@Aloshi oh, good catch. Granted this if-statement shouldn't evaluate to true much at all, but we should still be correct about it.  A couple ideas:

1) Just save the `last_enabled` field. Worst case it clobbers a just-written value with an almost identical value.

```
if counter.last_enabled is None:
    counter.last_enabled = timezone.now()
    counter.save(update_fields=["last_enabled"])
```

2) Use a transaction (overkill?):

```
if counter.last_enabled is None:
    # Update the counter within a transaction. Refetch the counter using 
    # select_for_update so the row is locked until the transaction is complete.
    with transaction.atomic():
        counter = ActivityCounter.objects.select_for_update().get(id=counter.id)
        # Check last_enabled again in case it was just updated 
        if counter.last_enabled is None:
            counter.last_enabled = timezone.now()
            counter.save()
```

Thoughts?

Reply:
Cool, i'll go w method 1